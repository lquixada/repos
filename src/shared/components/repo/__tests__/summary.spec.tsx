import {mount} from 'enzyme'
import React from 'react'

import {Summary} from '../summary'

describe('<Summary />', () => {
  it('matches the snapshot', () => {
    const repo = {
      description: 'Github Repos',
      forks_count: 5,
      html_url: 'http://github-repos/',
      license: {
        name: 'MIT',
      },
      name: 'Github Repos',
      open_issues_count: 5,
      stargazers_count: 5,
      subscribers_count: 5,
    }
    const component = mount(<Summary owner='owner1' repo={repo} />)
    expect(component).toMatchSnapshot()
  })
})
