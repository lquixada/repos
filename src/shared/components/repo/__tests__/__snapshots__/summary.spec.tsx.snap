// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Summary /> matches the snapshot 1`] = `
<Summary
  owner="owner1"
  repo={
    Object {
      "description": "Github Repos",
      "forks_count": 5,
      "html_url": "http://github-repos/",
      "license": Object {
        "name": "MIT",
      },
      "name": "Github Repos",
      "open_issues_count": 5,
      "stargazers_count": 5,
      "subscribers_count": 5,
    }
  }
>
  <styled.ul>
    <ul
      className="sc-bdVaJa dElRfG"
    >
      <styled.li>
        <li
          className="sc-bwzfXH coPMTf"
        >
          <styled.a
            href="https://github.com/owner1/Github Repos/stargazers"
          >
            <a
              className="sc-htpNat iIQOQQ"
              href="https://github.com/owner1/Github Repos/stargazers"
              rel="noopener"
              target="_blank"
            >
              <StarIcon>
                <svg
                  aria-hidden={true}
                  className="octicon octicon-star"
                  height={16}
                  viewBox="0 0 14 16"
                  width={14}
                >
                  <path
                    d="M14 6l-4.9-.64L7 1 4.9 5.36 0 6l3.6 3.26L2.67 14 7 11.67 11.33 14l-.93-4.74z"
                    fillRule="evenodd"
                  />
                </svg>
              </StarIcon>
              5
               stars
            </a>
          </styled.a>
        </li>
      </styled.li>
      <styled.li>
        <li
          className="sc-bwzfXH coPMTf"
        >
          <styled.a
            href="https://github.com/owner1/Github Repos/watchers"
          >
            <a
              className="sc-htpNat iIQOQQ"
              href="https://github.com/owner1/Github Repos/watchers"
              rel="noopener"
              target="_blank"
            >
              <EyeIcon>
                <svg
                  aria-hidden={true}
                  className="octicon octicon-eye"
                  height={16}
                  viewBox="0 0 16 16"
                  width={16}
                >
                  <path
                    d="M8.06 2C3 2 0 8 0 8s3 6 8.06 6C13 14 16 8 16 8s-3-6-7.94-6zM8 12c-2.2 0-4-1.78-4-4 0-2.2 1.8-4 4-4 2.22 0 4 1.8 4 4 0 2.22-1.78 4-4 4zm2-4c0 1.11-.89 2-2 2-1.11 0-2-.89-2-2 0-1.11.89-2 2-2 1.11 0 2 .89 2 2z"
                    fillRule="evenodd"
                  />
                </svg>
              </EyeIcon>
              5
               watchers
            </a>
          </styled.a>
        </li>
      </styled.li>
      <styled.li>
        <li
          className="sc-bwzfXH coPMTf"
        >
          <styled.a
            href="https://github.com/owner1/Github Repos/network"
          >
            <a
              className="sc-htpNat iIQOQQ"
              href="https://github.com/owner1/Github Repos/network"
              rel="noopener"
              target="_blank"
            >
              <RepoForkedIcon>
                <svg
                  aria-hidden={true}
                  className="octicon octicon-repo-forked"
                  height={16}
                  viewBox="0 0 10 16"
                  width={10}
                >
                  <path
                    d="M8 1a1.993 1.993 0 0 0-1 3.72V6L5 8 3 6V4.72A1.993 1.993 0 0 0 2 1a1.993 1.993 0 0 0-1 3.72V6.5l3 3v1.78A1.993 1.993 0 0 0 5 15a1.993 1.993 0 0 0 1-3.72V9.5l3-3V4.72A1.993 1.993 0 0 0 8 1zM2 4.2C1.34 4.2.8 3.65.8 3c0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2zm3 10c-.66 0-1.2-.55-1.2-1.2 0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2zm3-10c-.66 0-1.2-.55-1.2-1.2 0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2z"
                    fillRule="evenodd"
                  />
                </svg>
              </RepoForkedIcon>
              5
               forks
            </a>
          </styled.a>
        </li>
      </styled.li>
      <styled.li>
        <li
          className="sc-bwzfXH coPMTf"
        >
          <styled.a
            href="https://github.com/owner1/Github Repos/issues"
          >
            <a
              className="sc-htpNat iIQOQQ"
              href="https://github.com/owner1/Github Repos/issues"
              rel="noopener"
              target="_blank"
            >
              <IssueOpenedIcon>
                <svg
                  aria-hidden={true}
                  className="octicon octicon-issue-opened"
                  height={16}
                  viewBox="0 0 14 16"
                  width={14}
                >
                  <path
                    d="M7 2.3c3.14 0 5.7 2.56 5.7 5.7s-2.56 5.7-5.7 5.7A5.71 5.71 0 0 1 1.3 8c0-3.14 2.56-5.7 5.7-5.7zM7 1C3.14 1 0 4.14 0 8s3.14 7 7 7 7-3.14 7-7-3.14-7-7-7zm1 3H6v5h2V4zm0 6H6v2h2v-2z"
                    fillRule="evenodd"
                  />
                </svg>
              </IssueOpenedIcon>
              5
               issues
            </a>
          </styled.a>
        </li>
      </styled.li>
      <styled.li>
        <li
          className="sc-bwzfXH coPMTf"
        >
          <styled.span>
            <span
              className="sc-bxivhb jsZpbe"
            >
              <LawIcon>
                <svg
                  aria-hidden={true}
                  className="octicon octicon-law"
                  height={16}
                  viewBox="0 0 14 16"
                  width={14}
                >
                  <path
                    d="M7 4c-.83 0-1.5-.67-1.5-1.5S6.17 1 7 1s1.5.67 1.5 1.5S7.83 4 7 4zm7 6c0 1.11-.89 2-2 2h-1c-1.11 0-2-.89-2-2l2-4h-1c-.55 0-1-.45-1-1H8v8c.42 0 1 .45 1 1h1c.42 0 1 .45 1 1H3c0-.55.58-1 1-1h1c0-.55.58-1 1-1h.03L6 5H5c0 .55-.45 1-1 1H3l2 4c0 1.11-.89 2-2 2H2c-1.11 0-2-.89-2-2l2-4H1V5h3c0-.55.45-1 1-1h4c.55 0 1 .45 1 1h3v1h-1l2 4zM2.5 7L1 10h3L2.5 7zM13 10l-1.5-3-1.5 3h3z"
                    fillRule="evenodd"
                  />
                </svg>
              </LawIcon>
              MIT
            </span>
          </styled.span>
        </li>
      </styled.li>
    </ul>
  </styled.ul>
</Summary>
`;
